### PR TITLE
8339235: Fix indentation in build system

### PR DESCRIPTION
**Pull request by magicus**
The indentation in the build system should be two spaces for logical indents, and four spaces for broken lines, period.

I searched for files starting with an odd number of spaces, and fixed those.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339235](https://bugs.openjdk.org/browse/JDK-8339235): Fix indentation in build system (**Bug** - P4)

### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/20763/head:pull/20763` \
`$ git checkout pull/20763`

Update a local copy of the PR: \
`$ git checkout pull/20763` \
`$ git pull https://git.openjdk.org/jdk.git pull/20763/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 20763`

View PR using the GUI difftool: \
`$ git pr show -t 20763`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/20763.diff">https://git.openjdk.org/jdk/pull/20763.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/20763#issuecomment-2317093380)

### CODE REVIEW

File: make/CompileJavaModules.gmk [hunk 1]
```diff
@@ -75,12 +75,12 @@ $(JDK_OUTPUTDIR)/modules/%_zh_HK.properties: $(JDK_OUTPUTDIR)/modules/%_zh_TW.pr
 
 CreateHkTargets = \
     $(call FilterExcludedTranslations, \
-       $(patsubst $(TOPDIR)/src/%, $(JDK_OUTPUTDIR)/modules/%, \
-         $(subst /share/classes,, \
+      $(patsubst $(TOPDIR)/src/%, $(JDK_OUTPUTDIR)/modules/%, \
```
Comment (erikj79) [inline make/CompileJavaModules.gmk:78]: Isn't this a continuation?

File: make/autoconf/boot-jdk.m4 [hunk 2]
```diff
@@ -602,11 +602,12 @@ AC_DEFUN([BOOTJDK_SETUP_BUILD_JDK],
...
-       if test "x$with_build_jdk" != x; then
-         BUILD_JDK=$with_build_jdk
-         BUILD_JDK_FOUND=maybe
-         AC_MSG_NOTICE([Found potential Build JDK using configure arguments])
-       fi])
+      if test "x$with_build_jdk" != x; then
+        BUILD_JDK=$with_build_jdk
+        BUILD_JDK_FOUND=maybe
+        AC_MSG_NOTICE([Found potential Build JDK using configure arguments])
+      fi
```
Comment (erikj79) [inline make/autoconf/boot-jdk.m4:609]: And here?

**Review by erikj79 — COMMENTED** [review]
Thanks for fixing this! See two nits. Also for anyone wondering, we defined these conventions for the build system a long time ago: https://openjdk.org/groups/build/doc/code-conventions.html

File: make/CompileJavaModules.gmk (hunk 1 above)
Comment (magicus) [reply make/CompileJavaModules.gmk:78]: Yep. I missed that. Thanks!

File: make/autoconf/boot-jdk.m4 (hunk 2 above)
Comment (magicus) [reply make/autoconf/boot-jdk.m4:609]: Actually, I think this is correct. The block is logically indented inside the `([` and `])` start-of-block and end-of-block markers. Kind of the same reasons as all m4 macros (as created by `AC_DEFUN`) is indented two spaces, and not four.

We have precedence for this pattern elsewhere as well, e.g. at line 240 in this file.

**Review by erikj79 — APPROVED** [review]

**Pull request closed without merging**